### ISSUE TITLE
`rule-empty-line-before` doesn't work inside LESS detached rulesets

### ISSUE THREAD

**Issue by thybzi** [stylelint/stylelint#4301]
> Clearly describe the bug

We use LESS detached rulesets http://lesscss.org/features/#detached-rulesets-feature
and wish to require an empty line before every single rule inside the ruleset (as well as outside it)
But the `rule-empty-line-before` doesn't make any check inside the ruleset.

> Which rule, if any, is the bug related to?

`rule-empty-line-before`

> What CSS is needed to reproduce the bug?

```less
.mixin(@ruleset) {
    @ruleset();
}

div {
    .mixin({
        a { color: red; }
        span { line-height: 42px; }
    });
}```

> What stylelint configuration is needed to reproduce the bug?

```json
{
    "rules": {
        "rule-empty-line-before": [ "always", {
            "ignore": ["after-comment"]
        }]
    }
}
```

> Which version of stylelint are you using?

`11.0.0`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

grunt-stylelint 0.11.0 (which uses Node.js API)

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Yes, it's related to LESS detached rulesets: http://lesscss.org/features/#detached-rulesets-feature

> What did you expect to happen?

The following warnings to be flagged:

```shell
Expected empty line before rule (rule-empty-line-before)
```

> What actually happened (e.g. what warnings or errors did you get)?

No warnings were flagged.

**Comment by evilebottnawi** [comment]
I think it is expected (https://github.com/stylelint/stylelint/blob/master/lib/rules/rule-empty-line-before/index.js#L54), we avoid checking non standard css, for non standard css like less/sass/scss we recommend to create plugin (like `stylelint-scss`)

**Comment by thybzi** [comment]
okay, so this plugin doesn't exists for less? @evilebottnawi

**Comment by evilebottnawi** [comment]
@thybzi it seems no, you can be the first

**Comment by thybzi** [comment]
if we create this plugin, how should we publish it to be available for all users?

**Comment by hudochenkov** [comment]
https://stylelint.io/user-guide/plugins

**Comment by evilebottnawi** [comment]
Feel free to add the plugin in the list when it will be implemented
